src/viam/sdk/components/movementsensor/movementsensor.hpp

```diff
@@ -0,0 +1,210 @@
+/// @file components/movementsensor/movementsensor.hpp
+///
+/// @brief Defines a `MovementSensor` component.
+#pragma once
+
+#include <string>
+
+#include <viam/api/component/movementsensor/v1/movementsensor.pb.h>
+
+#include <viam/sdk/common/linear_algebra.hpp>
+#include <viam/sdk/common/proto_type.hpp>
+#include <viam/sdk/common/utils.hpp>
+#include <viam/sdk/config/resource.hpp>
+#include <viam/sdk/registry/registry.hpp>
+#include <viam/sdk/resource/resource_manager.hpp>
+#include <viam/sdk/spatialmath/orientation.hpp>
+
+namespace viam {
+namespace sdk {
+
+/// @defgroup MovementSensor Classes related to the MovementSensor component.
+
+/// @class MovementSensorRegistration
+/// @brief Defines a `ResourceRegistration` for the `MovementSensor` component.
+/// @ingroup MovementSensor
+class MovementSensorRegistration : public ResourceRegistration {
+   public:
+    MovementSensorRegistration(const google::protobuf::ServiceDescriptor* service_descriptor);
+    std::shared_ptr<ResourceServer> create_resource_server(
+        std::shared_ptr<ResourceManager> manager) override;
+    std::shared_ptr<Resource> create_rpc_client(std::string name,
+                                                std::shared_ptr<grpc::Channel> chan) override;
+};
+
+/// @class MovementSensor movementsensor.hpp "components/movementsensor/movementsensor.hpp"
+/// @brief TODO
+/// @ingroup MovementSensor
+///
+/// This acts as an abstract base class to be inherited from by any drivers representing
+/// specific movementsensor implementations. This class cannot be used on its own.
+class MovementSensor : public Component {
+   public:
+    /// @struct compassheading
+    /// @brief TODO.
+    /// TODO(pre-merge) I am dubious that this needs a struct... Though it could be nice to have
+    /// helper functions like compassheadings.east()..?
+    struct compassheading {
+        /// A number from 0-359 where 0 is North, 90 is East, 180 is South, and 270 is   West
+        double value;
+    };
+
+    // @struct orientation
+    // @brief MovementSensor is the only user of viam::common::v1::Orientation right now.
+    // Unfortunately, that is not the same as viam::app::v1::Orientation (which is what spatialmath
+    // uses).
+    struct orientation {
+        double o_x, o_y, o_z, theta;
+        // TODO(pre-merge) move the to/from protos in here. Nicer that way
```

Comment:
Maybe? I saw you do this for some other methods? I kept the current way with everything in the component..

Not sure what the plan is